src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts

```diff
@@ -61,12 +63,27 @@ export class AdminSearchPageComponent {
         this.statusMessageService.showWarningToast('No results found.');
         this.instructors = [];
         this.students = [];
-      } else {
-        this.instructors = resp.instructors;
-        this.students = resp.students;
-        this.hideAllInstructorsLinks();
-        this.hideAllStudentsLinks();
+        return;
       }
+
+      this.instructors = resp.instructors;
+      this.students = resp.students;
+      this.hideAllInstructorsLinks();
+      this.hideAllStudentsLinks();
+
+      // prompt user to use more specific terms if search results limit reached
+      if (this.instructors.length >= this.maxResultsToShow && this.students.length >= this.maxResultsToShow) {
+        this.statusMessageService.showWarningToast(`${this.maxResultsToShow} student results and
+            ${this.maxResultsToShow} instructor results have been shown on this page but there may be more
+            results not shown. Consider searching with more specific terms.`);
+      } else if (this.instructors.length >= this.maxResultsToShow) {
+        this.statusMessageService.showWarningToast(`${this.maxResultsToShow} instructor results have been shown
+            on this page but there may be more results not shown. Consider searching with more specific terms.`);
+      } else if (this.students.length >= this.maxResultsToShow) {
+        this.statusMessageService.showWarningToast(`${this.maxResultsToShow} student results have been shown
+            on this page but there may be more results not shown. Consider searching with more specific terms.`);
```

Comment:
Too many repetitions here. Try to refactor such that only the front part (before "have been shown") is generated on demand.